lib/matplotlib/backend_bases.py

```diff
@@ -3058,14 +3058,50 @@ def press_pan(self, event):
         if (event.button not in [MouseButton.LEFT, MouseButton.RIGHT]
                 or event.x is None or event.y is None):
             return
-        axes = [a for a in self.canvas.figure.get_axes()
-                if a.in_axes(event) and a.get_navigate() and a.can_pan()]
-        if not axes:
+
+        # get all relevant axes and group them by patch-visibility and zorder
+        all_axes = defaultdict(lambda: defaultdict(list))
+        for a in self.canvas.figure.get_axes():
+            if a.in_axes(event) and a.get_navigate() and a.can_pan():
+                all_axes[a.patch.get_visible()][a.zorder].append(a)
```

Comment:
Not commenting on the overall strategy; just on the current implementation...

The use of defaultdict and of `all_axes[True]` / `all_axes[False]` below is not so easy to follow.  I would just do something like (untested, and perhaps further simplifications are possible) `visible_axes_by_z = {}; invisible_axes_by_z = {}` then
```python
for ax in get_axes():
    if ax.get_visible(): visible_axes_by_z.setdefault(ax.zorder, []).append(ax)
    else: invisible_axes_by_z.setdefault(ax.zorder, []).append(ax)
```
(one problem with defaultdict is that *any* `__getitem__` could be inserting a new key into the dict, and it's hard to follow which ones do so; using setdefault makes this explicit.)

Reply:
OK, I can change that... I thought that using a defaultdict should be fine in here since it stays inside the function and using it makes the code more compact. (but I agree that if you're not used to them they can be somewhat confusing...)

Reply:
@anntzer I've now updated the implementation to use explicit `dict.setdefault(key, []).append()` and `dict.get(key, [])` instead of a `defaultdict`